File: app/src/main/java/vertretunggut/app/niklas/vertretungsplan/RepPlanAdapter.java
package vertretunggut.app.niklas.vertretungsplan;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.ArrayList;

/**
 * Created by nwuensche on 30.09.16.
 */
public class RepPlanAdapter extends ArrayAdapter<RepPlanLine> {

    public RepPlanAdapter(Context context, ArrayList<RepPlanLine> lines) {
        super(context, 0, lines);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        RepPlanLine line = getItem(position);

        if(convertView == null){
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.list_row, parent, false);
        }

        TextView hour = (TextView) convertView.findViewById(R.id.hour);
        TextView teacher = (TextView) convertView.findViewById(R.id.teacher);
        TextView subject = (TextView) convertView.findViewById(R.id.subject);
        TextView room = (TextView) convertView.findViewById(R.id.room);
        TextView schoolClass = (TextView) convertView.findViewById(R.id.schoolClass);
        TextView type = (TextView) convertView.findViewById(R.id.type);
        TextView message = (TextView) convertView.findViewById(R.id.message);

        hour.setText(line.getHour());
        teacher.setText(line.getTeacher());
        subject.setText(line.getSubject());
        room.setText(line.getRoom());
        schoolClass.setText(line.getSchoolClass());
        type.setText(line.getType());
        message.setText(line.getMessage());

        return convertView;
    }

}

File: app/src/main/java/vertretunggut/app/niklas/vertretungsplan/RepPlanAdapter.kt
package vertretunggut.app.niklas.vertretungsplan

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import java.util.ArrayList

/**
 * Created by nwuensche on 30.09.16.
 */
class RepPlanAdapter(context: Context, lines: ArrayList<RepPlanLine>) : ArrayAdapter<RepPlanLine>(context, 0, lines) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val convertView = convertView ?: LayoutInflater.from(context).inflate(R.layout.list_row, parent, false)

        val line = getItem(position)

        convertView!!.run {
            findViewById<TextView>(R.id.hour).text = line!!.hour
            findViewById<TextView>(R.id.teacher).text = line.teacher
            findViewById<TextView>(R.id.subject).text = line.subject
            findViewById<TextView>(R.id.room).text = line.room
            findViewById<TextView>(R.id.schoolClass).text = line.schoolClass
            findViewById<TextView>(R.id.type).text = line.type
            findViewById<TextView>(R.id.message).text = line.message
        }
        return convertView
    }

}
